behind_ingensec.rb

```diff
@@ -0,0 +1,285 @@
+##
+# This module requires Metasploit: http://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Auxiliary
+
+  def initialize(info = {})
+    super(update_info(info,
+      'Name'           => 'Behind BinarySec/IngenSecurity',
+      'Version' => '$Release: 1.0.1',
+      'Description'    => %q{
+        This module can be useful if you need to test
+        the security of your server and your website
+        behind BinarySec/IngenSec by discovering the real IP address.
+      },
+      'Author'         => [
+        'mekhalleh',
+        'RAMELLA Sébastien <sebastien.ramella[at]Pirates.RE>'
+      ],
+      'References' => [
+        ['URL', 'https://www.ingensecurity.com/fct_apercu.html']
+      ],
+      'License'        => MSF_LICENSE
+    ))
+
+    register_options(
+      [
+        OptString.new('HOSTNAME', [true, 'The hostname to find real IP', 'www.reunionnaisdumonde.com']),
+        OptString.new('URIPATH', [true, 'The URI path (for custom comparison)', '/']),
+        OptInt.new('RPORT', [true, 'The target port (for custom comparison)', 443]),
+        OptBool.new('SSL', [true, 'Negotiate SSL/TLS for outgoing connections (for custom comparison)', true]),
+        OptString.new('Proxies', [false, 'A proxy chain of format type:host:port[,type:host:port][...]']),
+        OptInt.new('THREADS', [false, 'Threads for DNS enumeration', 1]),
+        OptPath.new('WORDLIST', [true, 'Wordlist of subdomains', ::File.join(Msf::Config.data_directory, 'wordlists', 'namelist.txt')])
+      ])
+  end
+
+  def do_check_tcp_port(ip, port, proxies)
+    begin
+      sock = Rex::Socket::Tcp.create(
+        'PeerHost' => ip,
+        'PeerPort' => port,
+        'Proxies'  => proxies
+      )
+    rescue ::Rex::ConnectionRefused, Rex::ConnectionError
+      return false
+    end
+    sock.close
+    return true
+  end
+
+  def do_grab_domain_ip_history(hostname, proxies)
+    begin
+      cli = Rex::Proto::Http::Client.new('www.prepostseo.com', 443, {}, true, nil, proxies)
+      cli.connect
+
+      request = cli.request_cgi({
+        'uri'    => '/domain-ip-history-checker',
+        'method' => 'POST',
+        'data'   => "url=#{hostname}&submit=Check+Reverse+Ip+Domains"
+      })
+      response = cli.send_recv(request)
+      cli.close
+
+    rescue ::Rex::ConnectionError, Errno::ECONNREFUSED, Errno::ETIMEDOUT
+      print_error('HTTP connection failed to PrePost SEO website.')
+      return false
+    end
+
+    html  = response.get_html_document
+    table = html.css('table.table').first
+    rows  = table.css('tr')
+
+    arIps = []
+    rows.each_with_index.map do | row, index |
+      row = /(\d*\.\d*\.\d*\.\d*)/.match(row.css('td').map(&:text).to_s)
+      unless row.nil?
+        arIps.push(row)
+      end
+    end
+
+    if arIps.empty?
```

Comment:
Please no camel-case. There are enough languages which hurt the eyes out there. As mentioned above, convention is snake-case.